src/python/WMCore/MicroService/MSTransferor/MSTransferor.py

```diff
@@ -676,3 +720,58 @@ def _diskPNNs(self, pnnList):
             else:
                 diskPNNs.add(pnn)
         return diskPNNs
+
+    def updateSites(self, rec):
+        """
+        Update sites API provides asynchronous update of site list information
+        :param rec: JSON payload with the following data structures: {'workflow': <wflow name>}
+        :return: either empty list (no errors) or list of errors
+        """
+        # preserve provided payload to local file system
+        wflowName = rec['workflow']
+        status = self.updateStorage(wflowName)
+        if status == 'ok':
+            return []
+        err = MSTransferorStorageError(rec)
+        self.logger.error(err)
+        return [err.error()]
+
+    def updateStorage(self, wflowName):
+        """
+        Save workflow data to persistent storage
+        :param wflowName: name of the workflow
+        :return: status of this operation
+        """
+        try:
+            fname = '{}/{}'.format(self.storage, wflowName)
+            with open(fname, 'w', encoding="utf-8"):
+                # we perform touch operation on file system, i.e. create empty file
+                self.logger.info("for workflow %s add entry to persistent storage", wflowName)
+                os.utime(fname, None)
```

Comment:
Given that we are persistent user information in the Disk, let us add some basic data sanitization rule to have protection against misuse.
I would suggest to validate this workflow name against the `requestName` Lexicon: https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/Lexicon.py#L499

Reply:
ok